CMakeLists.txt

```diff
@@ -1,69 +1,145 @@
-cmake_minimum_required(VERSION 3.21)
+cmake_minimum_required(VERSION 3.24)
 
 project(
-  NESO-Particles
+  NesoParticles
```

Comment:
Is "NESO-Particles" bad as a name? Maybe "NESOParticles" - I don't think we use "NesoParticles" anywhere except in NESO-Spack to fit the naming conventions.

Reply:
Thanks - I'll rebase off dev and change the name back - not sure why I changed it in the first place. 
Re. clang-format - Mine is newer (v15), what might work is if you do 
`clang-format --dump-config --style=llvm > .clang-format` 
I hope (maybe) that that config file would then give the same output across versions. If you send it me I can check.